aixplain/modules/agent/__init__.py

```diff
@@ -163,19 +166,39 @@ def run(
                 max_iterations=max_iterations,
                 output_format=output_format,
             )
-            if response["status"] == "FAILED":
+            if response["status"] == ResponseStatus.FAILED:
                 end = time.time()
                 response["elapsed_time"] = end - start
                 return response
             poll_url = response["url"]
             end = time.time()
-            response = self.sync_poll(poll_url, name=name, timeout=timeout, wait_time=wait_time)
-            return response
+            result = self.sync_poll(poll_url, name=name, timeout=timeout, wait_time=wait_time)
+            result_data= result.data
+            return AgentResponse(
```

Comment:
Some fields are still missing in data. `result_data` has the following fields:

['input', 'output', 'session_id', 'intermediate_steps', 'runTime', 'usedCredits', 'executionStats']

Please add all of them.